uefi-test-runner/src/proto/ata/pass_thru.rs

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use uefi::boot;
+use uefi::boot::{OpenProtocolAttributes, OpenProtocolParams};
+use uefi::proto::ata::pass_thru::AtaPassThru;
+use uefi::proto::ata::AtaRequestBuilder;
+
+pub fn test() {
+    info!("Running ATA PassThru tests");
+
+    assert!(try_find_drive());
+}
+
+fn try_find_drive() -> bool {
+    let ata_ctrl_handles = boot::find_handles::<AtaPassThru>().unwrap();
+    assert_eq!(ata_ctrl_handles.len(), 1);
+
+    for handle in ata_ctrl_handles {
+        let params = OpenProtocolParams {
+            handle,
+            agent: boot::image_handle(),
+            controller: None,
+        };
+        let ata_pt = unsafe {
+            // don't open exclusive! That would break other tests
+            boot::open_protocol::<AtaPassThru>(params, OpenProtocolAttributes::GetProtocol).unwrap()
```

Comment:
but why? They do not run concurrently

Reply:
No idea. Maybe it unloads a driver that's required for later tests when I open it exclusively.
Using `open_protocol_exclusive` here results in:

```
[PANIC]: panicked at uefi-test-runner/src/proto/device_path.rs:29:85:
called `Result::unwrap()` on an `Err` value: Error { status: INVALID_PARAMETER, data: () }
```